uber/config.py

```diff
@@ -698,27 +699,67 @@ def ADMIN_WRITE_ACCESS_SET(self):
 
     @cached_property
     def ADMIN_PAGES(self):
-        # Build a list of all site sections and their pages
-        public_site_sections = ['static_views', 'angular', 'public', 'staffing']
-        public_pages = []
+        public_site_sections, public_pages, pages = self.gettable_site_sections_and_pages
         site_sections = cherrypy.tree.apps[c.CHERRYPY_MOUNT_PATH].root
-        modules = {name: getattr(site_sections, name) for name in dir(site_sections) if not name.startswith('_')}
-        for module_name, module_root in modules.items():
-            method = getattr(site_sections, module_name)
-            if getattr(method, 'public', False):
-                public_site_sections.append(module_name)
-            else:
-                for name in dir(module_root):
-                    if not name.startswith('_'):
-                        method = getattr(module_root, name)
-                        if getattr(method, 'public', False):
-                            public_pages.append(module_name + "_" + name)
 
         return {
             section: [opt for opt in dir(getattr(site_sections, section))
                       if opt not in public_pages and not opt.startswith('_')]
             for section in dir(site_sections) if section not in public_site_sections and not section.startswith('_')
         }
+        
+    @request_cached_property
+    def SITE_MAP(self):
+        public_site_sections, public_pages, pages = self.gettable_site_sections_and_pages
+        
+        accessible_site_sections = {section: pages for section, pages in pages.items() 
+                                    if c.has_section_or_page_access(page_path=section, include_read_only=True)}
+        for section in accessible_site_sections:
+            accessible_site_sections[section] = [page for page in accessible_site_sections[section] 
+                                                 if c.has_section_or_page_access(page_path=page['path'], include_read_only=True)]
+            
+        return sorted(accessible_site_sections.items())
+    
+    @cached_property
+    def gettable_site_sections_and_pages(self):
```

Comment:
I don't have a strong opinion on it either way, but is there a reason why this property isn't capitalized like the rest of them?

Reply:
Oops. It didn't start off as a property! I'll rename it.